public/app/core/components/NestedFolderPicker/NestedFolderPicker.tsx

```diff
@@ -82,12 +82,23 @@ export function NestedFolderPicker({
   const [error] = useState<Error | undefined>(undefined); // TODO: error not populated anymore
   const lastSearchTimestamp = useRef<number>(0);
 
+  // Map the permission string union to enum value for compatibility
+  const permissionLevel = useMemo(() => {
```

Comment:
Can't we change the type downstream? Would that be too many changes?

Reply:
Yeah.... I wanted to, but too many changes, especially going into the Searcher classes that I wanted to avoid in this PR.

I know it's not great to introduce yet another deprecation in this PR, but I think it's quite a surface level one that's not _that_ bad.